### DISCUSSION TITLE
Rename "MerkleStreamer" to "MerkleLockup"?

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/v2-periphery#222] [General]
I can't see any use case for open-ended streaming for airdrops .. but still. WDYT about renaming `MerkleStreamer` to `MerkleLockup` to reflect the fact that the current implementation targets the `Lockup` release?

Cc @andreivladbrg @razgraf

**Comment by andreivladbrg** [comment]
tbh "MerkleLockup" doesn't offer enough information about the contract - you know that you "lockup" some funds but what are you going to do with them? -  imo "streamer" word is very important, as it indicates that the contract combines two concepts, merkle trees + streaming 

i suggest we either keep "MerkleStreamer" or rename it to "MerkleStreamerLockup"

**Reply by PaulRBerg** [reply]
Welp, what happens when we add support for LD .. streaming won't be the only supported feature in that case.

Fair enough, though; if the user wants to do a simple one-time airdrop, then using Sablier V2 may be overkill. And so the name `MerkleLockup` becomes self-defeating.

**Reply by PaulRBerg** [reply]
Let's stick with `MerkleStreamer`.
